pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -541,15 +542,18 @@ func (r *Replica) applySnapshot(
 	}
 	var ingestStats pebble.IngestOperationStats
 	if ingestStats, err =
-		r.store.engine.IngestExternalFilesWithStats(ctx, inSnap.SSTStorageScratch.SSTs()); err != nil {
+		// TODO: separate ingestions for log and statemachine engine. See:
+		//
```

Comment:
What does "ingestion" mean for the log engine in the snapshot application path? Does it boil down to a few (range-)deletes and point puts?

It's the "state" engine which should "ingest" larger SSTs, right?

Reply:
> It's the "state" engine which should "ingest" larger SSTs, right?

Yep, that's the bulk of the data. On the log engine, we're mostly doing an in-place log truncation and writing a new HardState. And since we'll put some random rangeID-unreplicated stuff on that engine we'll need to ClearRange that, too, I think, or handle it on a case by case basis depending on intended semantics.